src/brogue/Monsters.c

```diff
@@ -2460,6 +2460,82 @@ boolean monsterSummons(creature *monst, boolean alwaysUse) {
     return false;
 }
 
+/// @brief Checks if a creature status effect is negatable and optionally negates it
+/// @param monst The creature 
+/// @param theStatus The status effect
+/// @param negate True to actually negate the status. False to just check if it's negatable.
+/// @return True if the status is negatable
+boolean canNegateCreatureStatusEffect(creature *monst, enum statusEffects theStatus, boolean negate) {
+
+    if (!monst || (monst->info.flags & MONST_INVULNERABLE)) {
+        return false;
+    }
+
+    if (monst->status[theStatus] > 0 && statusEffectCatalog[theStatus].isNegatable) {
+        if (negate) {
+            monst->status[theStatus] = (monst == &player) ? statusEffectCatalog[theStatus].playerNegatedValue : 0;
+            if (theStatus == STATUS_DARKNESS && monst == &player) {
+                updateMinersLightRadius();
+                updateVision(true);
+            }
+        }
+        return true;
+    }
+
+    return false;
+}
+
+/// @brief Checks if a creature has any negatable status effects and optionally negates them
+/// @param monst The creature
+/// @param negate True to actually negate. False to just check.
+/// @return True if the creature has any negatable status effects
+boolean canNegateCreatureStatusEffects(creature *monst, boolean negate) {
```

Comment:
Note/idea: Overall I'm not loving this name, given it can have an effect too, though I can see why you factored these two things together as they share a lot of code.

One possible idea, if you felt strongly, is to rename this then wrap calls to this function with two others, `canNegateCreatureStatusEffects` and `negateCreatureStatusEffects`

I'm happy to merge as-is though

Reply:
I was struggling with the side effect myself. I have an update along the lines of that idea and it's ready to push but I noticed the code has a reference to MONST_REFLECT4 so I'll wait until #680 is merged, rebase, and then force push the change.